app/src/main/java/com/realityexpander/tasky/presentation/login_screen/LoginScreen.kt

```diff
@@ -57,113 +86,201 @@ fun LoginScreen(
+            }
+            loginState.statusMessage.getOrNull?.let { message ->
+                Text(text = message)
+                Spacer(modifier = Modifier.extraSmallHeight())
+            }
 
-        // STATUS //////////////////////////////////////////
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .weight(1f)
+                    .background(color = MaterialTheme.colors.surface)
+                    .padding(bottom = DP.large)
+            ) {
+                this@col1.AnimatedVisibility(
+                    visible = !isKeyboardOpen,
+                    enter = fadeIn() + slideInVertically(
+                        initialOffsetY = { it }
+                    ),
+                    exit = fadeOut(),
+                    modifier = Modifier
+                        .background(color = MaterialTheme.colors.surface)
+                        .align(alignment = Alignment.BottomCenter)
+                ) {
+                    // • REGISTER TEXT BUTTON
+                    Text(
+                        text = UiText.Res(R.string.login_not_a_member_sign_up).get,
+                        style = MaterialTheme.typography.body1,
+                        color = MaterialTheme.colors.primaryVariant,
+                        modifier = Modifier
+                            .align(alignment = Alignment.BottomCenter)
+                            .clickable(onClick = {
+                                navigateToRegister()
+                            })
+                    )
+                }
+            }
+        }
+    }
+}
 
-        loginState.errorMessage.getOrNull()?.let { errorMessage ->
-            Text(
-                text = "Error: $errorMessage",
-                color = Color.Red,
+@Composable
+@Preview(
+    showBackground = true,
+    uiMode = Configuration.UI_MODE_NIGHT_YES,
+)
+fun LoginScreenPreview() {
+    TaskyTheme {
+        androidx.compose.material.Surface {
+            LoginScreen(
+                navigator = EmptyDestinationsNavigator,
+                viewModel = LoginViewModel(
+                    authRepository = AuthRepositoryFakeImpl(
+                        authApi = AuthApiFakeImpl(),
+                        authDao = AuthDaoFakeImpl(),
+                        validateUsername = ValidateUsername(),
+                        validateEmail = ValidateEmailImpl(),
+                        validatePassword = ValidatePassword(),
+                    ),
+                    validateEmail = ValidateEmailImpl(),
+                    validatePassword = ValidatePassword(),
+                    savedStateHandle = SavedStateHandle().apply {
+                        // For Live Preview
+                        set("email", "chris@demo.com")
+                        set("password", "123456Aa")
+                        set("confirmPassword", "123456Aa")
+                    }
+                )
             )
```

Comment:
I know I did this in some videos as well, but I recently started keeping Compose screens ViewModel-free because of this issue that you always need to initialize a big ViewModel. Instead, you can just pass the screen state as a parameter and `onAction: (LoginScreenAction) -> Unit` to it, which is much more intuitive to create.

Reply:
The above code was for preview purposes only.

Im not sure I understand what change you are suggesting.

Where is the viewModel instantiated and held? 
How do events/state get sent from the viewModel to the Compose Screen?
Wouldn't the ViewModel then contain a reference to the Compose Screen, instead of the other way around? Shouldnt the view depend on the viewModel, not the other way around?

Do you have an example I could look at that illustrates your suggestion?

I find all this ping-pong stuff to be somewhat silly over-engineering, IMHO. But it seems to be the way apps are made on android, so I am attempting to follow that pattern.

Reply:
Yeah, something like that, just that if you have one UI state for the whole screen, so you end up with something like this:
```
@Composable 
fun LoginScreenContent(
    state: LoginState,
    sendEvent: (LoginEvent) -> Unit
) {
    // ...
}
```

Using normal navigation, I instantiated the ViewModel in the `composable` block of the `NavHost`. If you use Compose Nav Destinations, your approach seems to be a convenient alternative.

Reply:
Sounds good